lib/block-supports/elements.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Elements styles block support.
+ *
+ * @package gutenberg
+ */
+
+/**
+ * Render out the duotone stylesheet and SVG.
+ *
+ * @param  string $block_content Rendered block content.
+ * @param  array  $block         Block object.
+ * @return string                Filtered block content.
+ */
+function gutenberg_render_elements_support( $block_content, $block ) {
+	$block_type = WP_Block_Type_Registry::get_instance()->get_registered( $block['blockName'] );
+	$link_color = _wp_array_get( $block['attrs'], array( 'style', 'elements', 'link', 'color', 'text' ), null );
+
+	/*
+	* For now we only care about link color.
+	* This code in the future when we have a public API
+	* should take advantage of WP_Theme_JSON::compute_style_properties
+	* and work for any element and style.
+	*/
+	if ( null === $link_color ) {
+		return $block_content;
+	}
+	$class_name = 'wp-block-elements-container-' . uniqid();
+
+	if ( strpos( $link_color, 'var:preset|color|' ) !== false ) {
+		// Get the name from the string and add proper styles.
+		$index_to_splice = strrpos( $link_color, '|' ) + 1;
+		$link_color_name = substr( $link_color, $index_to_splice );
+		$link_color      = "var(--wp--preset--color--$link_color_name)";
+	}
+
+	$style      = implode(
+		"\n",
+		array(
+			'<style>',
+			".$class_name a{",
```

Comment:
Do you think it's important that instead of `a` here we get the element selector from the existing [elements array](https://github.com/WordPress/gutenberg/blob/trunk/lib/class-wp-theme-json.php#L252)? I'm leaning towards using it.

Reply:
The elements array is not exposed is private information from that class in /lib/class-wp-theme-json.php.
I don't think we should expose the elements array. It is not something very useful to be part of the public API and elements were just merged this week. If we want to expose something I think we should expose a function that given a styles object compiles it to CSS (compute_style_properties) and in that case, we don't need to expose anything. But The public API is not yet decided, I would prefer if for now we don't expose anything and just use the "a" and then we expose some function to compile styles when our API is ready.